files/usr/share/cinnamon/applets/power@cinnamon.org/applet.js

```diff
@@ -67,52 +39,69 @@ function deviceLevelToString(level) {
     }
 }
 
-function deviceTypeToString(type) {
-    switch (type) {
-        case UPDeviceType.AC_POWER:
+function deviceKindToString(kind) {
+    // By default we use .replaceAll('-', ' ') and .capitalize() to make the
+    // UPowerGlib.kind_to_string() result more friendly, but we also handle a
+    // few special cases here where that does't provide an optimal result.
+    switch (kind) {
+        case UPDeviceKind.LINE_POWER:
             return _("AC adapter");
-        case UPDeviceType.BATTERY:
+        case UPDeviceKind.BATTERY:
             return _("Laptop battery");
-        case UPDeviceType.UPS:
+        case UPDeviceKind.UPS:
             return _("UPS");
-        case UPDeviceType.MONITOR:
-            return _("Monitor");
-        case UPDeviceType.MOUSE:
-            return _("Mouse");
-        case UPDeviceType.KEYBOARD:
-            return _("Keyboard");
-        case UPDeviceType.PDA:
+        case UPDeviceKind.PDA:
             return _("PDA");
-        case UPDeviceType.PHONE:
+        case UPDeviceKind.PHONE:
             return _("Cell phone");
-        case UPDeviceType.MEDIA_PLAYER:
-            return _("Media player");
-        case UPDeviceType.TABLET:
-            return _("Tablet");
-        case UPDeviceType.COMPUTER:
-            return _("Computer");
-        default:
-            return _("Unknown");
+        case UPDeviceKind.BLUETOOTH_GENERIC:
+            return _("Bluetooth device");
+        default: {
+            try {
+                return _(
+                    UPDevice
+                        .kind_to_string(kind)
+                        .replaceAll('-', ' ')
+                        .capitalize()
+                );
```

Comment:
This would end up returning untranslated strings. I think we should keep the existing translated device types, then do this to handle any newly added device types.